src/middlewares/notification.middleware.js

```diff
@@ -0,0 +1,19 @@
+import PreferenceService from '../services/preference.service';
+import ResponseService from '../services/response.service';
+
+const checkIfOptionExists = async (req, res, next) => {
+  const prefs = await PreferenceService.findPreferences({ userId: req.userData.id });
+  const { dataValues } = prefs;
+
+  const isValueSet = Object.keys(dataValues)
+    .filter((key) => key === req.preferenceMode)
```

Comment:
A user can only have one preference so there is no need to filter and map. Use sequelize find method to get only one preference from the database then do your check.

Reply:
He can have them both. In DB Bernard suggested we create two columns inApp and email preference. The values are booleans.  When a user submit an update then I need to find the row to update(filter) then check its value. if it is not equal to value from body(map) that when I continue.